lua/entities/gmod_wire_expression2/core/e2lib.lua

```diff
@@ -439,89 +439,135 @@ end
 
 do
-	local extensions = {}
+	E2Lib.extensions = { status = {}, list = {} }
+	local sortExtensionsList
 
-	if not sql.TableExists("wire_expression2_extensions") then
-		sql.Query("CREATE TABLE IF NOT EXISTS wire_expression2_extensions (name varchar(255), enabled tinyint)")
-		sql.Query("CREATE UNIQUE INDEX name ON wire_expression2_extensions(name)")
-	end
-
-	function extensions.GetStatus(name, default)
-		local value = sql.QueryValue(string.format("SELECT enabled FROM wire_expression2_extensions WHERE (name = %s)", sql.SQLStr(name)))
-		value = value and value ~= "0" or default
-		return value
+	if not sql.TableExists( "wire_expression2_extensions" ) then
+		sql.Query( "CREATE TABLE IF NOT EXISTS wire_expression2_extensions (name varchar(255), enabled tinyint)" )
+		sql.Query( "CREATE UNIQUE INDEX name ON wire_expression2_extensions(name)" )
+	else
+		local q = sql.Query( "SELECT * FROM wire_expression2_extensions" )
+		if q then
+			for _, row in ipairs( q ) do
+				E2Lib.extensions.status[ row.name ] = row.enabled == "1" and true or false
+				E2Lib.extensions.list[ #E2Lib.extensions.list + 1 ] = row.name
+				sortExtensionsList = true
+			end
+		end
 	end
 
-	function extensions.SetStatus(name, status)
-		sql.Query(string.format("REPLACE INTO wire_expression2_extensions (name, enabled) VALUES (%s, %d)", sql.SQLStr(name), status and 1 or 0))
+	function E2Lib.SetExtensionStatus( name, status )
+		E2Lib.extensions.status[ name ] = status
+		sql.Query( string.format( "REPLACE INTO wire_expression2_extensions (name, enabled) VALUES (%s, %d)", sql.SQLStr( name ), status and 1 or 0 ) )
 	end
 
-	function E2Lib.RegisterExtension(name, default)
-		local status = extensions.GetStatus(name, default)
-		if not status then Error("Skipping disabled E2 extension '" .. name .. "'. To enable, run 'wire_expression2_extension_enable " .. name .. "'\n") end
+	--[[ useless yet
+	function E2Lib.GetExtensionStatus( name )
+		return E2Lib.extensions.status[ name ] or false
 	end
-
-	concommand.Add("wire_expression2_extension_enable", function(ply, cmd, args)
-		if ply:IsValid() then
-			if not ply:IsSuperAdmin() and not game.SinglePlayer() then return end
+	]]
+	
+	function E2Lib.GetExtensions( niceList )
+		if sortExtensionsList then
+			table.sort( E2Lib.extensions.list, function( a, b ) return a < b end )
+			sortExtensionsList = nil
 		end
-
-		if extensions.GetStatus(args[1]) then
-			if ply:IsValid() then
-				ply:PrintMessage(2, "Extension '" .. args[1] .. "' is already enabled. Did you remember to reload Expression 2 using the console command 'wire_expression2_reload'?")
-			else
-				print("Extension '" .. args[1] .. "' is already enabled. Did you remember to reload Expression 2 using the console command 'wire_expression2_reload'?")
-			end
-			return
+		
+		if not niceList then return E2Lib.extensions.list end
+		
+		local l = 0
+		local t = {}
+		for _, n in ipairs( E2Lib.extensions.list ) do if n:len() > l then l = n:len() end end
```

Comment:
Use "\t" instead of this if you're going to print things. Also `#str` is always faster than `str:len()`.